Definitions/Key

```diff
@@ -1,26 +1,30 @@
 class Key:
-    def __init__(self, symbol, startNote, isMajor):
-        this.symbol = symbol
-        this.startNote = startNote%12
-        this.isMajor = isMajor
+    def __init__(self, symbol, tonic, isMajor):
+        self.symbol = symbol
+        self.tonic = tonic%12
+        self.isMajor = isMajor
+        self.scale = this.getScale()
     
     def getScale(self):
-        scale = list()
-        counter = 0
-        if this.isMajor:
+        scale = dict{tonic:"1"}
+        counter = self.tonic
+        degreeCounter = 2
+        if self.isMajor:
             pattern = [2,2,1,2,2,2,1]
             for step in pattern:
                 note = (counter + step)%12
-                scale.append(note)
+                scale[note] = degreeCounter
+                degreeCounter += 1
         else:
             pattern = [2,1,2,2,1,2,2]
             for step in pattern:
                 note = (counter + step)%12
-                scale.append(note)
+                scale[note] = degreeCounter
+                degreeCounter += 1
         return scale
     
     def getScaleDegree(self, note):
-        if note%12 in self.getScale:
-            return scale.index(note%12) + 1
+        if note%12 in self.getScale().keys():
```

Comment:
You don't need to extract the keys from the dict that is returned. The `in` keyword will automatically check for for keys in a dict.
e.g.
```
1 in {1: "some value"} #True
```